draft-age-masque-connect-ip.md

```diff
@@ -260,107 +261,74 @@ the peer of the assignment.
-
-IP Address:
-
-: IP address of the advertised route. If the IP Version field has value 4, the
-IP Address field SHALL have a length of 32 bits. If the IP Version field has
-value 6, the IP Address field SHALL have a length of 128 bits.
+: IP Version of this range. MUST be either 4 or 6.
 
-IP Prefix Length:
+Start IP Address and End IP Address:
 
-: The number of bits in the IP Address that are used to define the prefix of
-the advertised route. This MUST be lesser or equal to the length of the IP
-Address field, in bits. If the prefix length is equal to the length of the IP
-Address, this route only allows sending packets to a single destination
-address. If the prefix length is less than the length of the IP address, this
-route allows sending packets to any destination address that falls within the
-prefix.
+: Inclusive start and end IP address of the advertised range. If the IP Version
+field has value 4, these fields SHALL have a length of 32 bits. If the IP
+Version field has value 6, these fields SHALL have a length of 128 bits. The
+Start IP Address MUST be strictly lesser than the End IP Address.
 
 IP Protocol:
 
-: The Internet Protocol Number for traffic that can be sent to this prefix. If
+: The Internet Protocol Number for traffic that can be sent to this range. If
 the value is 0, all protocols are allowed.
 
 Upon receiving the ROUTE_ADVERTISEMENT capsule, an endpoint MAY start routing
-IP packets in that prefix to its peer.
-
-If an endpoint receives multiple ROUTE_ADVERTISEMENT capsules, all of the
-advertised routes can be used. For example, multiple ROUTE_ADVERTISEMENT
-capsules are necessary to provide routing to both IPv4 and IPv6 hosts. Routes
-are removed using ROUTE_WITHDRAWAL capsules.
+IP packets in these ranges to its peer.
 
-### ROUTE_WITHDRAWAL Capsule
+Each ROUTE_ADVERTISEMENT contains the full list of address ranges. If multiple
+ROUTE_ADVERTISEMENT capsules are sent in one direction, each
+ROUTE_ADVERTISEMENT capsule supersedes prior ones. In other words, if a given
+address range was present in a prior capsule but the most recently received
+ROUTE_ADVERTISEMENT capsule does not contain it, the receiver will consider
+that range withdrawn.
 
-The ROUTE_WITHDRAWAL capsule (see {{iana-types}} for the value of the capsule
-type) allows an endpoint to communicate to its peer that it is not willing to
-route traffic to a given prefix.
+In order to guarantee the absence of overlap between ranges using the same IP
+Protocol, the ranges are ordered. If an IP Address Range A precedes an IP
+address range B in the same ROUTE_ADVERTISEMENT capsule, they MUST follow these
+requirements:
 
-~~~
-ROUTE_WITHDRAWAL Capsule {
-  Type (i) = ROUTE_WITHDRAWAL,
-  Length (i),
-  IP Version (8),
-  IP Address (32..128),
-  IP Prefix Length (8),
-  IP Protocol (8),
-}
-~~~
-{: #route-withdraw-format title="ROUTE_WITHDRAWAL Capsule Format"}
+* IP Version of A MUST be lesser or equal than IP Version of B
```

Comment:
This seems actually rather arbitrary.

Reply:
Agreed, but I think it just needs to be an arbitrary choice if you want a consistent order.